src/wagtail_vector_index/storage/pgvector/provider.py

```diff
@@ -90,6 +91,23 @@ def get_similar_documents(
             embedding = pgvector_embedding.embedding
             yield embedding.to_document()
 
+    async def aget_similar_documents(
+        self, query_vector, *, limit: int = 5
+    ) -> AsyncGenerator[Document, None]:
+        async for pgvector_embedding in (
+            self._get_queryset()
```

Comment:
Is it worth factoring out this queryset in to a common helper method? Seems like this might be only one extra function, so not too concerned about the extra complexity.